lib/msal-node/src/client/PublicClientApplication.ts

```diff
@@ -60,4 +60,32 @@ export class PublicClientApplication extends ClientApplication {
             throw e;
         }
     }
+
+    /**
+     * Acquires tokens with password grant by exchanging client applications username and password for credentials
+     *
+     * The latest OAuth 2.0 Security Best Current Practice disallows the password grant entirely.
+     * More details on this recommendation at https://tools.ietf.org/html/draft-ietf-oauth-security-topics-13#section-3.4
+     * Microsoft's documentation and recommendations are at:
+     * https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#usernamepassword
+     *
+     * @param request
+     */
+    async acquireTokenByUsernamePassword(request: UsernamePasswordRequest): Promise<AuthenticationResult> {
```

Comment:
Is there any way to throw an error if this API is used in a non-testing environment? Otherwise we want to more explicitly discourage developers from using it.

Reply:
I do not think we should throw. Just heavily discourage in docs.